src/luasql.c

```diff
@@ -131,3 +131,33 @@ LUASQL_API void luasql_set_info (lua_State *L) {
 	lua_pushliteral (L, "LuaSQL 2.3.0");
 	lua_settable (L, -3);
 }
+
+/*
+** Pulls an optional string value from the table at idx
+*/
+LUASQL_API const char* luasql_table_optstring(lua_State *L, int idx, const char* name, const char* def) {
+	const char* res = NULL;
+
+	lua_pushstring(L, name);
+	lua_gettable(L, idx);
+
+	res = lua_tostring(L, -1);
+	lua_pop(L, 1);
+
+	return (res != NULL) ? res : def;
+}
+
+/*
+** Pulls an optional number value from the table at idx
+*/
+LUASQL_API lua_Number luasql_table_optnumber(lua_State *L, int idx, const char* name, lua_Number def) {
+	lua_Number res = 0;
+
+	lua_pushstring(L, name);
+	lua_gettable(L, idx);

```

Comment:
Hi

> The last person who I saw keeping Lua 5.0 compatibility in any library
> was @tomasguisasola [2]. Since he's maintaining LuaSQL drivers for his
> uses in production, he'd be the best person to ask. Tomás, do you
> still need 5.0 compatibility for anything?
In fact I didn't _needed_ compatibility.  It was just a challenge
to let the code compatible to all Lua 5.X versions at the time I
managed to make the code compatible to Lua 5.2 :-)

Regards,
Tomás Guisasola

On 2016-07-04 16:40, Hisham Muhammad wrote:
> In src/luasql.c [1]:
> 
>> + lua_gettable(L, idx);
>> +
>> + res = lua_tostring(L, -1);
>> + lua_pop(L, 1);
>> +
>> + return (res != NULL) ? res : def;
>> +}
>> +
>> +/*
>> +** Pulls an optional number value from the table at idx
>> +*/
>> +LUASQL_API lua_Number luasql_table_optnumber(lua_State *L, int idx,
> const char* name, lua_Number def) {
>> + lua_Number res = 0;
>> +
>> + lua_pushstring(L, name);
>> + lua_gettable(L, idx);
> 
> The last person who I saw keeping Lua 5.0 compatibility in any library
> was @tomasguisasola [2]. Since he's maintaining LuaSQL drivers for his
> uses in production, he'd be the best person to ask. Tomás, do you
> still need 5.0 compatibility for anything?
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [3], or mute the
> thread [4].
> 
> 	*
> 
> Links:
> ------
> [1] 
> https://github.com/keplerproject/luasql/pull/54#discussion_r69487576
> [2] https://github.com/tomasguisasola
> [3]
> https://github.com/keplerproject/luasql/pull/54/files/b4afdfb93567f8ffaa268ed63115bb9672a106aa#r69487576
> [4]
> https://github.com/notifications/unsubscribe/AAIA7Q9s-8A6oIFVU_2kSr4SXNoXH8AYks5qSWGngaJpZM4JEQdV